xbmc/peripherals/bus/PeripheralBus.h

```diff
@@ -54,7 +54,7 @@ namespace PERIPHERALS
     /*!
      * @return True if this bus needs to be polled for changes, false if this bus performs updates via callbacks
      */
-    bool NeedsPolling(void) const { return m_bNeedsPolling; }
+    bool NeedsPolling(void) const { CSingleLock lock(m_critSection); return m_bNeedsPolling; }
```

Comment:
Same as a3b9caf it's to protect it from concurrent writes. Afaik just because it's bool doesn't give it any guaranties like that. That's what std::atomic is for.